pkg/scheduler/framework/runtime/framework.go

```diff
@@ -1016,62 +1016,56 @@ func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.Cy
 	allNodePluginScores := make([]framework.NodePluginScores, len(nodes))
 	numPlugins := len(f.scorePlugins) - state.SkipScorePlugins.Len()
 	plugins := make([]framework.ScorePlugin, 0, numPlugins)
-	pluginToNodeScores := make(map[string]framework.NodeScoreList, numPlugins)
+	pluginToNodeScores := make([]framework.NodeScoreList, numPlugins)
 	for _, pl := range f.scorePlugins {
 		if state.SkipScorePlugins.Has(pl.Name()) {
 			continue
 		}
 		plugins = append(plugins, pl)
-		pluginToNodeScores[pl.Name()] = make(framework.NodeScoreList, len(nodes))
 	}
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	errCh := parallelize.NewErrorChannel()
 
-	if len(plugins) > 0 {
-		logger := klog.FromContext(ctx)
-		logger = klog.LoggerWithName(logger, "Score")
-		// TODO(knelasevero): Remove duplicated keys from log entry calls
-		// When contextualized logging hits GA
-		// https://github.com/kubernetes/kubernetes/issues/111672
-		logger = klog.LoggerWithValues(logger, "pod", klog.KObj(pod))
-		// Run Score method for each node in parallel.
-		f.Parallelizer().Until(ctx, len(nodes), func(index int) {
-			nodeName := nodes[index].Name
+	logger := klog.FromContext(ctx)
+	logger = klog.LoggerWithName(logger, "Score")
+	// TODO(knelasevero): Remove duplicated keys from log entry calls
+	// When contextualized logging hits GA
+	// https://github.com/kubernetes/kubernetes/issues/111672
+	logger = klog.LoggerWithValues(logger, "pod", klog.KObj(pod))
+	// Run Score method for each node in parallel.
+	f.Parallelizer().Until(ctx, len(plugins), func(i int) {
```

Comment:
Previously, we were parallelising over the nodes, now we are paralellising over the number of plugins, which is much much lower.

The benchmark presented in the PR only goes up to 100 nodes, which is not good enough justification for this change.

Please revert.

Reply:
+1 to revert this since we reached the code freeze.